acceptance/tests/direct_puppet/supports_utf8.rb

```diff
@@ -16,13 +19,15 @@
       #   so we can't easily use expect_failure here
       skip_test 'PUP-6217'
     end
-    tmp_file[agent.hostname] = agent.tmpfile(tmp_environment)
+    tmp_file[agent_to_fqdn(agent)] = agent.tmpfile(tmp_environment)
   end
 
   teardown do
     step 'clean out produced resources' do
       agents.each do |agent|
-        on(agent, "rm #{tmp_file[agent.hostname]}", :accept_all_exit_codes => true)
+        if tmp_file.has_key?(agent_to_fqdn(agent)) && !tmp_file[agent_to_fqdn(agent)].empty?
+          on(agent, "rm -f '#{tmp_file[agent_to_fqdn(agent)]}'")
```

Comment:
Previously, this would be OK with `rm` failing to remove the file (either because it wasn't there to remove, or because it couldn't remove it). With `-f`, `rm` shouldn't complain if the file isn't there, but will still exit non-zero if it can't remove a file that does exist. Is this the desired behavior?